### PR TITLE
Print service changes with credential status update

### PR DESCRIPTION
**Pull request by MuralitharanK**

### CODE REVIEW

File: src/main/java/io/mosip/print/service/impl/PrintServiceImpl.java [hunk 1]
```diff
@@ -354,6 +410,7 @@ private Map<String, byte[]> getDocuments(String credential, String credentialTyp
 				eventName = EventName.UPDATE.toString();
 				eventType = EventType.BUSINESS.toString();
 			} else {
+				printStatusUpdate(requestId, CredentialStatusConstant.ERROR.name(), null);
```
Comment (vishwa-vyom) [inline src/main/java/io/mosip/print/service/impl/PrintServiceImpl.java:413]: We are not passing the error details in the event for debugging later?

File: src/main/java/io/mosip/print/service/impl/PrintServiceImpl.java [hunk 2]
```diff
@@ -232,7 +286,7 @@ public boolean generateCard(EventModel eventModel) {
 	private Map<String, byte[]> getDocuments(String credential, String credentialType, String encryptionPin,
 			String requestId,
 			String cardType,
-			boolean isPasswordProtected) {
+			boolean isPasswordProtected, String refId, String registrationId) {
```
Comment (vishwa-vyom) [inline src/main/java/io/mosip/print/service/impl/PrintServiceImpl.java:289]: why is refId passed and not getting used ?

File: src/main/java/io/mosip/print/service/impl/PrintServiceImpl.java [hunk 3]
```diff
@@ -548,80 +615,88 @@ private void setTemplateAttributes(String jsonString, Map<String, Object> attrib
...
+            if (object != null) {
+                try {
+                    obj = new JSONParser().parse(object.toString());
+                } catch (Exception e) {
+                    obj = object;
+                }
+            }
+            if (obj instanceof JSONArray) {
+                JsonValue[] jsonValues = JsonUtil.mapJsonNodeToJavaObject(JsonValue.class, (JSONArray) obj);
+                uinCardPd = uinCardPd.concat(getFormattedPasswordAttribute(getParameter(jsonValues, templateLang)).substring(0, 4));
```
Comment (vishwa-vyom) [inline src/main/java/io/mosip/print/service/impl/PrintServiceImpl.java:676]: any reason we are hardcoding as 4 here ?

Comment (vishwa-vyom) [reply src/main/java/io/mosip/print/service/impl/PrintServiceImpl.java:676]: Also why this is not followed for object else if case ?

**Review by vishwa-vyom — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 4]
```diff
@@ -229,11 +241,6 @@
 			<artifactId>jackson-datatype-jsr310</artifactId>
 			<version>2.12.0</version>
 		</dependency>
-		<dependency>
-			<groupId>info.weboftrust</groupId>
-			<artifactId>ld-signatures-java</artifactId>
-			<version>0.8.0</version>
-		</dependency>
```
[the remark this answers is not in the document]
Comment (MuralitharanK) [reply pom.xml:236]: > why is this dependency removed ?

We need this dependency but dont need this as a direct dependency.. 
Its a coming as transitive dependency from "vcverifier"

File: src/main/java/io/mosip/print/service/impl/PrintServiceImpl.java [hunk 5]
```diff
@@ -185,42 +194,87 @@ public class PrintServiceImpl implements PrintService{
...
+
+	@Override
+    public boolean generateCard(EventModel eventModel) {
+        boolean isPrinted = false;
+        try {
+            printStatusUpdate(eventModel.getEvent().getTransactionId(), CredentialStatusConstant.RECEIVED.name(), null);
+            String credential = getCredential(eventModel);
+            String decodedCredential = decryptCredential(credential);
+            printLogger.debug("vc is printed security valuation.... : {}", decodedCredential);
+            if (!hasPrintCredentialVerified(eventModel, decodedCredential)) return false;
```
[the remark this answers is not in the document]
Comment (MuralitharanK) [reply src/main/java/io/mosip/print/service/impl/PrintServiceImpl.java]: done.

File: src/main/java/io/mosip/print/service/impl/PrintServiceImpl.java [hunk 6]
```diff
@@ -373,6 +430,16 @@ private Map<String, byte[]> getDocuments(String credential, String credentialTyp
 		return byteMap;
 	}
 
+    private String getDatashareUrl(byte[] data) throws IOException, DataShareException, ApiNotAccessibleException {
+        DataShare dataShare = dataShareUtil.getDataShare(data, policyId, partnerId);
+        return dataShare.getUrl().replace("http://", "https://");
```
[the remark this answers is not in the document]
Comment (MuralitharanK) [reply src/main/java/io/mosip/print/service/impl/PrintServiceImpl.java]: removed the protocol replacement, added for testing purpose.

File: src/main/java/io/mosip/print/service/impl/PrintServiceImpl.java [hunk 7]
```diff
@@ -373,6 +430,16 @@ private Map<String, byte[]> getDocuments(String credential, String credentialTyp
...
+    }
+
+    private String getRid(Object id) {
+        return id.toString().split("/credentials/")[1];
```
[the remark this answers is not in the document]
Comment (MuralitharanK) [reply src/main/java/io/mosip/print/service/impl/PrintServiceImpl.java]: Unused method removed.

File: src/main/java/io/mosip/print/service/impl/PrintServiceImpl.java (hunk 2 above)
Comment (MuralitharanK) [reply src/main/java/io/mosip/print/service/impl/PrintServiceImpl.java]: removed the unused arguments.

File: src/main/java/io/mosip/print/service/impl/PrintServiceImpl.java (hunk 3 above)
Comment (MuralitharanK) [reply src/main/java/io/mosip/print/service/impl/PrintServiceImpl.java]: taken care.

**Review by vishwa-vyom — APPROVED** [review]

**Pull request merged**